### PR TITLE
refactor(googlemaps): throw warnings plugin_not_installed

### PR DESCRIPTION
**Pull request by ramonornela**

### CODE REVIEW

File: src/plugins/googlemaps.ts [hunk 1]
```diff
@@ -1,13 +1,6 @@
...
+import { Cordova, CordovaInstance, Plugin, InstanceProperty, getPlugin, pluginWarn } from './plugin';
 import { Observable } from 'rxjs/Observable';
 
-
-/**
- * @private
- * Created by Ibrahim on 3/29/2016.
- */
-declare var plugin: any;
-
```
Comment (ramonornela) [inline src/plugins/googlemaps.ts]: @ihadeed review this deletion, attention with copyright 😉 

Thanks

**Comment by ihadeed** [comment]
Sorry I meant `@CordovaFunctionOverride` not `CordovaOverride`

**Comment by ramonornela** [comment]
New decorator available in 2.0.3 I am remembering added by @mlynch https://github.com/driftyco/ionic-native/commit/f60d08b7a4110d7e183a36df5f0dadf412726ca5, later fix reports

Thanks

**Comment by ihadeed** [comment]
@ramonornela sorry just had a look at the decorator again, it's not what we need here. But I was thinking we can create a decorator that will let us make sure that the plugin exists before running the function.

**Comment by ramonornela** [comment]
@ihadeed this decorator works with observable that make break bc to methods returns an promise

Maybe best adjust @Cordova and @CordovaInstance accept Promise

```
@CordovaInstance({
    promiseIndex: 0, // optional if not info add push
    promise: (marker: any) {
      if (marker) {
        Promise.resolve(new GoogleMapsMarker(marker));
      } else {
        Promise.reject();
      }
    }
  })
addMarker(options: GoogleMapsMarkerOptions): Promise<GoogleMapsMarker> 
```

Thanks

**Comment by ihadeed** [comment]
I think it's good enough to just have the check happen in the constructor of `GoogleMap` class and maybe the `GeoCoder` too.

**Comment by ramonornela** [comment]
Maybe not throw warning but adjust to not break in browser 😉

**Comment by ihadeed** [comment]
Sure we can do something like `if(!this._objectInstance) return;`

**Comment by ramonornela** [comment]
@ihadeed Adjust PR, thanks by report 😉

**Comment by ihadeed** [comment]
Looks good. Let's revisit this later and add utility functions in plugin.ts that return back `Promise.reject({ error: 'plugin_not_installed' });` and that observable that returns an error. So we don't have to rewrite the same thing many times.

**Comment by ramonornela** [comment]
I think that this rule should applied also to called @Cordova() and @CordovaInstance() ```Promise.reject() and Observer.error()``` but in 2.2 to not generate break in 2.1.*

Maybe create a branch release 2.2 to merge improvements.

Thanks

**Pull request merged**